nautobot/utilities/forms/forms.py

```diff
@@ -324,17 +326,27 @@ def capitalize(field):
 
     def _get_lookup_field_choices(self):
         """Get choices for lookup_fields i.e filterset parameters without a lookup expr"""
+        from nautobot.extras.filters.mixins import RelationshipFilter  # Avoid circular import
+
         filterset_without_lookup = (
-            (name, field.label or self.capitalize(field.field_name))
+            (
+                name,
+                field.label
+                or (isinstance(field, RelationshipFilter) and field.relationship.get_label(side=field.side))
```

Comment:
Same here, personally I'd prefer a more verbose but less "magical" way of setting this value.